lang/cs/Org.Apache.REEF.Wake/Remote/AuxillaryStreamingFunctions.cs

```diff
@@ -0,0 +1,275 @@
+                    return -1;
+                }
+
+                totalBytesRead += bytesRead;
+            }
+
+            return totalBytesRead;
+        }
+
+        /// <summary>
+        /// Read bytes asynchronously from the stream.
+        /// </summary>
+        /// <param name="stream">Stream fromwhich to read</param>
+        /// <param name="bytesToRead">Number of bytes to read</param>
+        /// <param name="buffer">byte buffer to read in to</param>
+        /// <param name="token">the cancellation token</param>
+        /// <param name="offset">offset in the buffer where we should start writing</param>
+        /// <returns>The number of bytes read if success otherwise -1</returns>
+        internal static async Task<int> ReadBytesAsync(Stream stream, int bytesToRead, byte[] buffer, CancellationToken token, int offset = 0)
+        {
+            int totalBytesRead = 0;
+            while (totalBytesRead < bytesToRead)
+            {
+                int bytesRead = await stream.ReadAsync(buffer, totalBytesRead, bytesToRead - totalBytesRead, token);
+                if (bytesRead == 0)
+                {
+                    // Read timed out or connection was closed
+                    return -1;
+                }
+
+                totalBytesRead += bytesRead;
+            }
+
+            return bytesToRead;
+        }
+        
+        /// <summary>
+        /// Writes an integer to the stream
+        /// </summary>
+        /// <param name="obj">The integer to write</param>
+        /// <param name="stream">The stream to write to.</param>
+        public static void IntToStream(int obj, Stream stream)
+        {
+            stream.Write(BitConverter.GetBytes(obj), 0, sizeof(int));
+        }
+
+        /// <summary>
+        /// Reads an integer from the stream
+        /// </summary>
+        /// <param name="stream">The stream to write to.</param>
+        /// <returns>The read integer</returns>
+        public static int? StreamToInt(Stream stream)
+        {
+            byte[] intBytes = new byte[sizeof(int)];
+            int readBytes = ReadBytes(stream, sizeof (int), ref intBytes);
+
+            if (readBytes == -1)
+            {
+                return null;
+            }
+            return BitConverter.ToInt32(intBytes, 0);
+        }
+
+        /// <summary>
+        /// Writes an integer to the stream
+        /// </summary>
+        /// <param name="obj">The integer to write</param>
+        /// <param name="stream">The stream to write to.</param>
+        public static void LongToStream(long obj, Stream stream)
+        {
+            stream.Write(BitConverter.GetBytes(obj), 0, sizeof(long));
+        }
+
+        /// <summary>
+        /// Reads an integer from the stream
+        /// </summary>
+        /// <param name="stream">The stream to write to.</param>
+        /// <returns>The read integer</returns>
+        public static long? StreamToLong(Stream stream)
```

Comment:
@beysims, @jwang98052 can you comment on the use of `null` in C# in this case? My gut feeling would be to throw an `IOException` or such instead of returning `null`.